consensus/src/chained_bft/consensus_types/block.rs

```diff
@@ -320,10 +328,20 @@ where
     }
 
     pub fn height(&self) -> Height {
+        // Height:
+        // - should not exceed std::u64::MAX - 1 to ensure the parent check doesn't
+        // cause addition overflow.
+        // (Block::make_block)
+        assumed_postcondition!(self.height < std::u64::MAX);
```

Comment:
This is essentially assuming that the block chain will not grow in length (height) enough to exceed u64::MAX in the next million years at least, which seems very reasonable.